wasm/module.go

```diff
@@ -78,3 +80,61 @@ func DecodeModule(binary []byte) (*Module, error) {
 	}
 	return ret, nil
 }
+
+func (m *Module) GetFunctionNames() (map[uint32]string, error) {
+	namesec, ok := m.CustomSections["name"]
+	if !ok {
+		return nil, fmt.Errorf("'name' %w", ErrCustomSectionNotFound)
+	}
+
+	r := bytes.NewReader(namesec)
+	for {
+		id, err := r.ReadByte()
+		if err != nil {
+			return nil, fmt.Errorf("failed to read subsection ID: %w", err)
+		}
+
+		size, _, err := leb128.DecodeUint32(r)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read the size of subsection: %w", err)
```

Comment:
maybe add the subsection ID into the error message? I would use code suggestion, but not sure the numeric encoding norms (ex if hex or otherwise)